packages/fx-core/tests/plugins/resource/aad/unit/generateArmTemplates.test.ts

```diff
@@ -0,0 +1,209 @@
+
+    // Assert
+    const mockedSolutionDataContext = {
+      plugins: activeResourcePlugins,
+      "fx-resource-aad-app-for-teams": {},
+      "fx-resource-frontend-hosting": {
+        outputs: {
+          domain: "test_frontend_hosting_domain_url",
+        },
+      },
+      "fx-resource-simple-auth": {},
+    };
+    chai.assert.isTrue(result.isOk());
+    if (result.isOk()) {
+      const expectedResult = mockSolutionUpdateArmTemplates(
+        mockedSolutionDataContext,
+        result.value
+      );
+
+      const expectedBicepFileDirectory = path.join(__dirname, "expectedBicepFiles");
+      const expectedParameterFilePath = path.join(expectedBicepFileDirectory, "input_param.bicep");
+      chai.assert.strictEqual(
+        expectedResult.Orchestration.ParameterTemplate!.Content,
+        fs.readFileSync(expectedParameterFilePath, ConstantString.UTF8Encoding)
+      );
+      const expectedVariablesFilePath = path.join(
+        expectedBicepFileDirectory,
+        "tab",
+        "variables.bicep"
+      );
+      chai.assert.strictEqual(
+        expectedResult.Orchestration.VariableTemplate!.Content,
+        fs.readFileSync(expectedVariablesFilePath, ConstantString.UTF8Encoding)
+      );
+      const expectedParameterJsonFilePath = path.join(
+        expectedBicepFileDirectory,
+        "parameters.json"
+      );
+      chai.assert.strictEqual(
+        expectedResult.Orchestration.ParameterTemplate!.ParameterFile,
+        fs.readFileSync(expectedParameterJsonFilePath, ConstantString.UTF8Encoding)
+      );
+      chai.assert.isUndefined(expectedResult.Modules);
+      chai.assert.isUndefined(expectedResult.Orchestration.ModuleTemplate);
+      chai.assert.isUndefined(expectedResult.Orchestration.OutputTemplate);
+    }
+  });
+
+  it("generate arm templates: bot", async function () {
+    // Act
+    const activeResourcePlugins = [ResourcePlugins.Bot, ResourcePlugins.Aad];
+    const pluginContext: PluginContext = await TestHelper.pluginContext(new Map(), true, false);
+    pluginContext.projectSettings!.solutionSettings = {
+      name: "test_solution",
+      version: "1.0.0",
+      activeResourcePlugins: activeResourcePlugins,
+    } as AzureSolutionSettings;
+    const result = await AADPlugin.generateArmTemplates(pluginContext);
+
+    // Assert
+    const mockedSolutionDataContext = {
+      plugins: activeResourcePlugins,
+      "fx-resource-bot": {
+        outputs: {
+          domain: "test_bot_domain_url",
+        },
+      },
+      "fx-resource-aad-app-for-teams": {},
+    };
+    chai.assert.isTrue(result.isOk());
+    if (result.isOk()) {
+      const expectedResult = mockSolutionUpdateArmTemplates(
+        mockedSolutionDataContext,
+        result.value
+      );
+
+      const expectedBicepFileDirectory = path.join(__dirname, "expectedBicepFiles");
+      const expectedParameterFilePath = path.join(expectedBicepFileDirectory, "input_param.bicep");
```

Comment:
may only need to check the variable content? since other content is already covered by other test case

Reply:
Fixed.